src/lib/Parser.hs

```diff
@@ -260,6 +264,52 @@ topLet = do
         k = if v == mkName "eff" then EffectRowKind else TypeKind
         uTyKind = WithSrc (Just pos) $ UPrimExpr $ TCExpr k
 
+interfaceDef :: Parser [UDecl]
+interfaceDef = do
+  keyWord InterfaceKW
+  -- TODO: allow multi-parameter type classes
+  interfaceName <- upperName <|> symName
+  (oneAnnBinder, pos) <- withPos annBinder
+  keyWord WhereKW
+  record <- withSrc $ URecordTy <$> interfaceRecordFields ":"
+  let consName = mkInterfaceConsName interfaceName
+  let (varTypeName, oneAnnBinder') = case oneAnnBinder of
+        Bind v    -> (varName v, oneAnnBinder)
+        Ignore ty -> let genVar = mkName $ mkNoShadowingStr "gen" in
+            (genVar, Bind (genVar:>ty))
+  let tyCon = UConDef interfaceName $ toNest [oneAnnBinder']
```

Comment:
Instead of parsing an interface name and then an annotated binder, then putting it together, could you just use `tyConDef`? Seems like that would give you multi-parameter type classes for free, and you wouldn't have to manually assemble the `UConDef`.